tests/neg/i18588.scala

```diff
@@ -0,0 +1,22 @@
+class ROBox[+A](value: A) {
+  private var cached: A = value
+  def get: A = ROBox[A](value).cached
+}
+
+class Box[+A](value: A) {
+  private var cached: A = value // error
+  def get: A = cached
+
+  def put[AA >: A](value: AA): Unit = {
+    val box: Box[AA] = this
+    box.cached = value
+  }
+}
```

Comment:
Also add a valid variant of the read-write box as additional test case, to highlight the difference?
```scala
class BoxValid[+A](value: A, orig: A) {
  private var cached: A = value // ok
  def get: A = cached

  def reset(): Unit =
    cached = orig // ok: mutated through this prefix
}
```